pithos/plugins/journald_logging.py

```diff
@@ -79,7 +79,7 @@ def __init__(self, parent, settings):
         self.pithos_window = parent
         self.settings = settings
         self.set_resizable(False)
-        self.connect('response', self._on_response)
+        self.connect('delete-event', lambda *ignore: self.do_response(Gtk.ResponseType.CANCEL))
```

Comment:
Probably just want `self.response()`

Reply:
OK

Reply:
Well i guess I din't understand what you mean?
```self.connect('delete-event', lambda *ignore: self.do_response(self.response())```

Reply:
```python
self.connect('delete-event', lambda *ignore: self.response(Gtk.ResponseType.CANCEL))
 ```

Reply:
right on

Reply:
Nope that didn't work. I got:

```gtk_container_foreach: assertion 'GTK_IS_CONTAINER (container)' failed``

When I tried to reopen the window and a segfault and crash when I clicked on it.

Reply:
@TingPing the problem I'd guess is that ```self.response``` doesn't return True.

Reply:
```self.connect('delete-event', lambda *ignore: self.response(Gtk.ResponseType.CANCEL) or True)``` seems to work though.